Tutorial links don't work

Some links say "[link to Frame X]", but there isn't a link.

<img width="329" alt="screenshot 2016-07-02 16 53 13" src="https://cloud.githubusercontent.com/assets/403231/16540683/8969363e-4075-11e6-8add-2b347b3c5a13.png">

Tried to make one work with deeplinking and got a crash:

```
07-02 18:51:04.607 2761-3011/com.mozilla.hackathon.kiboko E/AndroidRuntime: FATAL EXCEPTION: ModernAsyncTask #2
                                                                            Process: com.mozilla.hackathon.kiboko, PID: 2761
                                                                            java.lang.RuntimeException: An error occurred while executing doInBackground()
                                                                                at android.support.v4.content.ModernAsyncTask$3.done(ModernAsyncTask.java:142)
                                                                                at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
                                                                                at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
                                                                                at java.util.concurrent.FutureTask.run(FutureTask.java:242)
                                                                                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                                                                                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                                                                                at java.lang.Thread.run(Thread.java:818)
                                                                             Caused by: java.lang.IllegalArgumentException: the bind value at index 1 is null
                                                                                at android.database.sqlite.SQLiteProgram.bindString(SQLiteProgram.java:164)
                                                                                at android.database.sqlite.SQLiteProgram.bindAllArgsAsStrings(SQLiteProgram.java:200)
                                                                                at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:47)
                                                                                at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1316)
                                                                                at android.database.sqlite.SQLiteDatabase.queryWithFactory(SQLiteDatabase.java:1163)
                                                                                at android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1034)
                                                                                at com.mozilla.hackathon.kiboko.utilities.SelectionBuilder.query(SelectionBuilder.java:170)
                                                                                at com.mozilla.hackathon.kiboko.provider.DsoProvider.query(DsoProvider.java:126)
                                                                                at android.content.ContentProvider.query(ContentProvider.java:1017)
                                                                                at android.content.ContentProvider$Transport.query(ContentProvider.java:238)
                                                                                at android.content.ContentResolver.query(ContentResolver.java:491)
                                                                                at android.support.v4.content.ContentResolverCompatJellybean.query(ContentResolverCompatJellybean.java:29)
                                                                                at android.support.v4.content.ContentResolverCompat$ContentResolverCompatImplJB.query(ContentResolverCompat.java:57)
                                                                                at android.support.v4.content.ContentResolverCompat.query(ContentResolverCompat.java:125)
                                                                                at android.support.v4.content.CursorLoader.loadInBackground(CursorLoader.java:59)
                                                                                at android.support.v4.content.CursorLoader.loadInBackground(CursorLoader.java:37)
                                                                                at android.support.v4.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:296)
                                                                                at android.support.v4.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:54)
                                                                                at android.support.v4.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:42)
                                                                                at android.support.v4.content.ModernAsyncTask$2.call(ModernAsyncTask.java:128)
                                                                                at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) 
                                                                                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) 
                                                                                at java.lang.Thread.run(Thread.java:818) 
```

--- a/app/src/main/java/com/mozilla/hackathon/kiboko/activities/IconQuizActivity.java
+++ b/app/src/main/java/com/mozilla/hackathon/kiboko/activities/IconQuizActivity.java
@@ -3,6 +3,7 @@ package com.mozilla.hackathon.kiboko.activities;
 
 import android.app.AlertDialog;
 import android.content.DialogInterface;
+import android.content.Intent;
 import android.database.Cursor;
 import android.os.Bundle;
 import android.support.annotation.IdRes;
@@ -14,7 +15,6 @@ import android.support.v7.app.ActionBar;
 import android.support.v7.app.AppCompatActivity;
 import android.view.KeyEvent;
 import android.view.MenuItem;
-import android.view.View;
 import android.widget.ImageView;
 import android.widget.TextView;
 
@@ -28,7 +28,6 @@ import com.mozilla.hackathon.kiboko.widgets.DSORadioGroup;
 import java.util.ArrayList;
 import java.util.List;
 
-import static com.mozilla.hackathon.kiboko.utilities.LogUtils.LOGI;
 import static com.mozilla.hackathon.kiboko.utilities.LogUtils.makeLogTag;
 
 /**
@@ -42,7 +41,7 @@ public class IconQuizActivity extends AppCompatActivity implements LoaderManager
     int question_id=0;
     Question currentQuestion;
     private String mQuiz;
-    TextView quizStepView;
+    TextView quizStepView, quizStepScore;
     CheckableLinearLayout choiceOptionA, choiceOptionB, choiceOptionC, choiceOptionD;
     ImageView imageOptionA,imageOptionB,imageOptionC,imageOptionD;
     DSORadioGroup quizGroup;
@@ -65,47 +64,18 @@ public class IconQuizActivity extends AppCompatActivity implements LoaderManager
         imageOptionD  = (ImageView)findViewById(R.id.answerd);
 
         quizStepView  = (TextView) findViewById(R.id.quizStepView);
+        quizStepScore = (TextView) findViewById(R.id.quizStepScore);
 
         quizGroup = (DSORadioGroup)findViewById(R.id.quizGroup);
 
         quizGroup.setOnCheckedChangeListener(new DSORadioGroup.OnCheckedChangeListener()
         {
-
             @Override
             public void onCheckedChanged(DSORadioGroup group, @IdRes int checkedId) {
                 if(checkedId != -1){
                     CheckableLinearLayout checkedView = (CheckableLinearLayout) findViewById(checkedId);
                     checkAnswer(checkedView);
                 }
-            }
-        });
-
-        //bootstrap content
-        choiceOptionA.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-//                checkAnswer(imageOptionA);
-            }
-        });
-
-        choiceOptionB.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                //checkAnswer(imageOptionB);
-            }
-        });
-
-        choiceOptionC.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                //checkAnswer(imageOptionC);
-            }
-        });
-
-        choiceOptionD.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-               // checkAnswer(imageOptionD);
             }
         });
 
@@ -126,11 +96,14 @@ public class IconQuizActivity extends AppCompatActivity implements LoaderManager
     @Override
     public boolean onKeyDown(int keyCode, KeyEvent event) { // this is override method
         if(keyCode == KeyEvent.KEYCODE_BACK){
-//            showExitConfirmDialog(); // call the function below
+            showExitConfirmDialog(); // call the function below
         }
         return super.onKeyDown(keyCode, event);
     }
 
+    /**
+     * Show exit confirm dialog
+     */
     public void showExitConfirmDialog(){ // just show an dialog
         AlertDialog.Builder dialog = new AlertDialog.Builder(this);
         dialog.setTitle("Exit Quiz?"); // set title
@@ -152,39 +125,41 @@ public class IconQuizActivity extends AppCompatActivity implements LoaderManager
         dialog.create().show();
     }
 
+    /**
+     * Check whether icon answer is correct {@link CheckableLinearLayout}
+     * @param checkedView selected view
+     * @return true/false
+     */
     public boolean checkAnswer(CheckableLinearLayout checkedView){
 
         boolean correct = false;
 
-        LOGI("yourans", currentQuestion.getANSWER() + " " + checkedView.getTag());
         if(currentQuestion.getANSWER().equals(checkedView.getTag()))
         {
             score++;
-            LOGI(TAG, "Your score" + score);
+            quizStepScore.setText(getString(R.string.quiz_template_score, score));
             correct = true;
         }else {
             correct = false;
         }
 
-        if(question_id < quizList.size() && correct){
+        if(question_id < quizList.size()){
             currentQuestion = quizList.get(question_id);
             setQuestionView(currentQuestion);
         }else{
-//            Intent intent = new Intent(IconQuizActivity.this, ResultActivity.class);
-//            Bundle b = new Bundle();
-//            b.putInt("score", score); //Your score
-//            intent.putExtras(b); //Put your score to your next Intent
-//            startActivity(intent);
-//            finish();
+            Intent intent = new Intent(IconQuizActivity.this, ResultActivity.class);
+            Bundle scoreBundle = new Bundle();
+            scoreBundle.putInt("score", score);
+            intent.putExtras(scoreBundle); // Send user score to your next Intent
+            startActivity(intent);
+            finish();
         }
-
         quizGroup.clearCheck();
-
         return correct;
     }
 
     /**
-     * Populate quiz question view
+     * Populate quiz question view {@link Question}
      */
     private void setQuestionView(Question question)
     {
@@ -197,8 +172,7 @@ public class IconQuizActivity extends AppCompatActivity implements LoaderManager
         imageOptionC.setImageDrawable(ContextCompat.getDrawable(this, Utils.getResId(this, question.getOPTIONC())));
         choiceOptionD.setTag(question.getOPTIOND());
         imageOptionD.setImageDrawable(ContextCompat.getDrawable(this, Utils.getResId(this, question.getOPTIOND())));
-
-        question_id++;
+        question_id ++;
     }
 
     @Override
@@ -212,11 +186,6 @@ public class IconQuizActivity extends AppCompatActivity implements LoaderManager
                 DsoContract.Quizes.KEY_OPTIONB,
                 DsoContract.Quizes.KEY_OPTIONC,
                 DsoContract.Quizes.KEY_OPTIOND};
-
-        // Read all data for quiz question Id
-        String selection = DsoContract.Quizes.KEY_ID + " =?";
-        String[] selectionArgs = new String[]{mQuiz};
-
         CursorLoader cursorLoader = new CursorLoader(IconQuizActivity.this,
                 DsoContract.Quizes.CONTENT_URI, // URI
                 projectionFields, // projection fields
@@ -240,7 +209,6 @@ public class IconQuizActivity extends AppCompatActivity implements LoaderManager
         }
 
         if(quizList.size() > 0){
-
             currentQuestion = quizList.get(question_id);
             setQuestionView(currentQuestion);
         }
@@ -253,7 +221,6 @@ public class IconQuizActivity extends AppCompatActivity implements LoaderManager
      */
     private Question loadQuizData(Cursor cursor) {
         // Get Question ID
-        int idIndex = cursor.getColumnIndex(DsoContract.Quizes.KEY_ID);
         int questionIndex = cursor.getColumnIndex(DsoContract.Quizes.KEY_QUESTION);
         int answerIndex = cursor.getColumnIndex(DsoContract.Quizes.KEY_ANSWER);
         int optionaIndex = cursor.getColumnIndex(DsoContract.Quizes.KEY_OPTIONA);

--- a/app/src/main/java/com/mozilla/hackathon/kiboko/activities/ResultActivity.java
+++ b/app/src/main/java/com/mozilla/hackathon/kiboko/activities/ResultActivity.java
@@ -1,10 +1,9 @@
 package com.mozilla.hackathon.kiboko.activities;
 
-import android.app.ActionBar;
-import android.app.Activity;
 import android.os.Bundle;
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatActivity;
 import android.view.MenuItem;
-import android.widget.RatingBar;
 import android.widget.TextView;
 
 import com.mozilla.hackathon.kiboko.R;
@@ -12,36 +11,33 @@ import com.mozilla.hackathon.kiboko.R;
 /**
  * Created by mwadime on 6/10/2016.
  */
-public class ResultActivity extends Activity {
+public class ResultActivity extends AppCompatActivity {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_result);
 
-        ActionBar ab = getActionBar();
+        ActionBar ab = getSupportActionBar();
         ab.setDisplayHomeAsUpEnabled(true);
-        //get rating bar object
-        RatingBar bar=(RatingBar)findViewById(R.id.ratingBar1);
-        bar.setNumStars(5);
-        bar.setStepSize(0.5f);
-        //get text view
-        TextView t=(TextView)findViewById(R.id.textResult);
+
+        TextView txtPoints = (TextView) findViewById(R.id.quizResult);
         //get score
         Bundle b = getIntent().getExtras();
         int score= b.getInt("score");
+
+        txtPoints.setText(getString(R.string.quiz_template_points, score));
         //display score
-        bar.setRating(score);
-        switch (score)
-        {
-            case 1:
-            case 2: t.setText("Opps, try again bro, keep learning");
-                break;
-            case 3:
-            case 4:t.setText("Nice Try");
-                break;
-            case 5:t.setText("Who are you? A student in JP???");
-                break;
-        }
+//        switch (score)
+//        {
+//            case 1:
+//            case 2: t.setText("Opps, try again bro, keep learning");
+//                break;
+//            case 3:
+//            case 4:t.setText("Nice Try");
+//                break;
+//            case 5:t.setText("Who are you? A student in JP???");
+//                break;
+//        }
     }
 
     @Override

--- a/app/src/main/java/com/mozilla/hackathon/kiboko/activities/TutorialSlideActivity.java
+++ b/app/src/main/java/com/mozilla/hackathon/kiboko/activities/TutorialSlideActivity.java
@@ -24,7 +24,6 @@ import com.google.gson.reflect.TypeToken;
 import com.mozilla.hackathon.kiboko.R;
 import com.mozilla.hackathon.kiboko.fragments.ScreenSlidePageFragment;
 import com.mozilla.hackathon.kiboko.models.Step;
-import com.mozilla.hackathon.kiboko.models.Tutorial;
 import com.mozilla.hackathon.kiboko.provider.DsoContract;
 
 import java.lang.reflect.Type;
@@ -70,7 +69,7 @@ public class TutorialSlideActivity extends DSOActivity implements LoaderManager.
     private String mTopic;
     TextView txtCaption;
 
-    private Tutorial tutorial;
+    private boolean fromLink;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
@@ -150,37 +149,11 @@ public class TutorialSlideActivity extends DSOActivity implements LoaderManager.
         String action = intent.getAction();
         String data = intent.getDataString();
         if (Intent.ACTION_VIEW.equals(action) && data != null) {
-            String tutorialId = data.substring(data.lastIndexOf("/") + 1);
+            String tutorialTag = data.substring(data.lastIndexOf("/") + 1);
+            mTopic = tutorialTag;
             Uri contentUri = DsoContract.Tutorials.CONTENT_URI.buildUpon()
-                    .appendPath(tutorialId).build();
-            showTutorial(contentUri);
+                    .appendPath(tutorialTag).build();
         }
-    }
-
-    private void showTutorial(Uri tutorialUri) {
-        LOGD("Tutorial Uri", tutorialUri.toString());
-
-        String[] projection = { DsoContract.Tutorials.TUTORIAL_ID,
-                DsoContract.Tutorials.TUTORIAL_TAG,
-                DsoContract.Tutorials.TUTORIAL_HEADER,
-                DsoContract.Tutorials.TUTORIAL_PHOTO_URL,
-                DsoContract.Tutorials.TUTORIAL_STEPS};
-
-        // Read all data for contactId
-        String selection = DsoContract.Tutorials.TUTORIAL_ID + " = ?";
-        String[] selectionArgs = new String[]{mTopic};
-
-//        Cursor cursor = getContentResolver().query(tutorialUri, projection, null, null, null);
-//
-//        if (cursor != null && cursor.moveToFirst()) {
-//            // always close the cursor
-//            cursor.close();
-//        } else {
-//            Toast toast = Toast.makeText(getApplicationContext(),
-//                    "No match for deep link " + tutorialUri.toString(),
-//                    Toast.LENGTH_SHORT);
-//            toast.show();
-//        }
     }
 
     @Override
@@ -204,8 +177,10 @@ public class TutorialSlideActivity extends DSOActivity implements LoaderManager.
                 DsoContract.Tutorials.TUTORIAL_PHOTO_URL,
                 DsoContract.Tutorials.TUTORIAL_STEPS };
 
+        String selection;
         // Read all data for contactId
-        String selection = DsoContract.Tutorials.TUTORIAL_TAG + " =?";
+        selection = DsoContract.Tutorials.TUTORIAL_TAG + " =?";
+
         String[] selectionArgs = new String[]{mTopic};
 
         CursorLoader cursorLoader = new CursorLoader(TutorialSlideActivity.this,
@@ -240,6 +215,8 @@ public class TutorialSlideActivity extends DSOActivity implements LoaderManager.
                     txtCaption.setText(getString(R.string.tutorial_template_step, 1, mPager.getAdapter().getCount()));
                 }
         }
+
+        cursor.close();
 
     }
 

--- a/app/src/main/java/com/mozilla/hackathon/kiboko/utilities/SelectionBuilder.java
+++ b/app/src/main/java/com/mozilla/hackathon/kiboko/utilities/SelectionBuilder.java
@@ -11,6 +11,7 @@ import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
+import static com.mozilla.hackathon.kiboko.utilities.LogUtils.LOGI;
 import static com.mozilla.hackathon.kiboko.utilities.LogUtils.LOGV;
 import static com.mozilla.hackathon.kiboko.utilities.LogUtils.makeLogTag;
 
@@ -165,7 +166,7 @@ public class SelectionBuilder {
                         String limit) {
         assertTable();
         if (columns != null) mapColumns(columns);
-        LOGV(TAG, "query(columns=" + Arrays.toString(columns)
+        LOGI(TAG, "query(columns=" + Arrays.toString(columns)
                 + ", distinct=" + distinct + ") " + this);
         return db.query(distinct, mTable, columns, getSelection(), getSelectionArgs(), mGroupBy,
                 mHaving, orderBy, limit);

--- a/app/src/main/res/layout/activity_icon_quiz.xml
+++ b/app/src/main/res/layout/activity_icon_quiz.xml
@@ -13,12 +13,22 @@
         android:gravity="center_vertical"
         android:layout_centerInParent="true"
         android:layout_centerHorizontal="true">
-
+        <TextView
+            android:layout_gravity="center_horizontal"
+            android:id="@+id/quizStepScore"
+            android:layout_width="wrap_content"
+            android:textColor="@color/colorTextPrimary"
+            android:gravity="center_horizontal"
+            android:layout_marginBottom="@dimen/activity_horizontal_margin"
+            android:layout_height="wrap_content"
+            android:text="@string/placeholder_quiz_score"
+            android:textAppearance="?android:attr/textAppearanceLarge" />
         <TextView
             android:layout_gravity="center_horizontal"
             android:id="@+id/quizStepView"
             android:layout_width="wrap_content"
             android:textColor="@color/colorTextPrimary"
+            android:textStyle="bold"
             android:gravity="center_horizontal"
             android:layout_marginBottom="@dimen/activity_horizontal_margin"
             android:layout_height="wrap_content"

--- a/app/src/main/res/layout/activity_result.xml
+++ b/app/src/main/res/layout/activity_result.xml
@@ -1,35 +1,38 @@
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:orientation="vertical"
     tools:context=".activities.ResultActivity"
-    android:background="@drawable/gray_border">
-
-    <LinearLayout
-        android:layout_width="wrap_content"
+    android:background="@color/colorPrimary">
+    <TextView
+        android:layout_margin="@dimen/activity_horizontal_margin"
+        android:layout_width="match_parent"
+        android:layout_gravity="center_horizontal"
+        android:gravity="center_horizontal"
         android:layout_height="wrap_content"
-        android:layout_alignParentBottom="true"
-        android:layout_alignParentLeft="true"
-        android:layout_alignParentRight="true"
-        android:layout_alignParentTop="true"
-        android:orientation="vertical" >
+        android:text="@string/game_over_text"
+        android:textColor="@color/colorTextPrimary"
+        android:textAppearance="?android:attr/textAppearanceLarge" />
+    <TextView
+        android:layout_margin="@dimen/activity_vertical_margin"
+        android:id="@+id/quizResult"
+        android:layout_gravity="center_horizontal"
+        android:gravity="center_horizontal"
+        android:textColor="@color/colorTextPrimary"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/quiz_template_points"
+        android:textAppearance="?android:attr/textAppearanceLarge" />
+    <Button
+        android:layout_margin="@dimen/activity_horizontal_margin"
+        android:layout_width="wrap_content"
+        android:background="#222222"
+        android:padding="@dimen/activity_horizontal_margin"
+        android:textColor="@color/colorTextPrimary"
+        android:layout_gravity="center_horizontal"
+        android:gravity="center_horizontal"
+        android:layout_height="wrap_content"
+        android:text="@string/play_again_text"/>
 
-
-        <RatingBar
-            android:id="@+id/ratingBar1"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
-            android:numStars="5"
-            android:stepSize="1.0"
-            android:rating="0.0"/>
-
-        <TextView
-            android:id="@+id/textResult"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_weight="0.08"
-            android:text="Large Text"
-            android:textAppearance="?android:attr/textAppearanceLarge" />
-    </LinearLayout>
-
-</RelativeLayout>
+</LinearLayout>

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -11,76 +11,13 @@
      <string name="title_template_step">Step <xliff:g id="step_number">%1$d</xliff:g>: Lorem
         Ipsum</string>
     <string name="tutorial_template_step">Step <xliff:g id="current_step_number">%1$d</xliff:g> of <xliff:g id="total_step_number">%2$d</xliff:g></string>
+    <string name="quiz_template_score">Score: <xliff:g id="current_score">%1$d</xliff:g></string>
     <string name="quiz_template_step">Question <xliff:g id="current_quiz_number">%1$d</xliff:g>/<xliff:g id="total_quiz_number">%2$d</xliff:g></string>
     <string name="action_previous">Previous</string>
     <string name="action_next">Next</string>
     <string name="action_finish">Finish</string>
 
     <string name="lorem_ipsum"><![CDATA[Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris et magna ut erat elementum cursus in quis ipsum. Nam faucibus ultrices eros, vel tempor leo semper sit amet. Duis magna quam, congue vitae interdum nec, condimentum ut lectus. Sed commodo venenatis eros a dignissim. Nunc adipiscing turpis nec magna venenatis at ultricies ipsum bibendum. Sed volutpat urna quis nisi vulputate egestas. Etiam eu dui a enim lacinia dapibus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas.\\n\\nPhasellus orci mi, eleifend nec congue id, volutpat at neque. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Integer semper velit vel leo rutrum sed fermentum elit porta.]]></string>
-
-    <!-- BEGIN Explore Message Card User Messages -->
-    <string name="explore_io_msgcards_ask_opt_in"
-        description="This message is displayed on the Explore I/O screen to ask users whether
-            they would like to opt in to receive messages regarding the conference."><![CDATA[
-        <b>Conference Messages</b><br/>
-       Would you like to opt-in to receive important messages about the conference? Messages are time based and may include promotional messages from partners.]]></string>
-
-    <string name="explore_io_msgcards_conf_creds_card"
-        description="Conference Credentials, shown the day before the event giving users general
-         event information."><![CDATA[
-        <b>Conference Credentials</b><br>
-        Each attendee will receive a non-transferable NFC enabled wristband this year. Your
-        wristband & I/O badge will allow you access to Keynote, Sessions, Sandbox, and After
-        Hours party. Remember to wear it at all times.
-    ]]>
-    </string>
-
-    <string name="explore_io_msgcards_after_hours_card"
-        description="This text is shown to a user when they are in the Explore I/O screen while at
-        the conference. It gives them information about an afternoon event on the schedule."><![CDATA[
-        <b>After Hours</b><br></br>
-        This is a party you don\'t want to miss! Attendees are invited to enjoy food, drinks,
-        music, games, and more during the evening of May 28. The conference badge and NFC wristband
-        are required for entrance.
-    ]]>
-    </string>
-
-    <string name="explore_io_msgcards_keynote_access_card"
-        description="This text is shown to a user when they are in the Explore I/O screen while at
-        the conference. It gives them information about the keynote event on the schedule."><![CDATA[
-        <b>Keynote Access</b><br>
-        Your wristband color denotes your keynote access.<br><br>
-        <b>Blue wristband = Keynote Access</b><br>
-        Line starts outside Moscone West on 4th Street and wraps around Minna Street. .<br><br>
-        <b>Gray wristband = Reserved Keynote Seating</b><br>
-        Line starts outside Moscone West on 4th Street.<br><br>
-        <b>Orange wristband = Playground</b><br>
-        Line starts outside Moscone West on Howard Street.
-    ]]>
-    </string>
-
-    <string name="explore_io_msgcards_wifi_feedback"
-        decscription="Question presented to users on a card in the Explore I/O screen if they have are using the conference Wifi.">Have connectivity feedback? Share it with us on social channels via the <b>#io15wifi</b> hashtag.</string>
-
-    <!-- This is used on the Explore I/O message cards to answer yes to a particular message -->
-    <string name="explore_io_msgcards_answer_yes">Sure, let\'s do it!</string>
-
-    <!-- This is used on the Explore I/O message cards to answer no to a particular message -->
-    <string name="explore_io_msgcards_answer_no">No thanks</string>
-
-    <!-- This string is presented to users when they attempt to click something that indicates they
-         can watch a video but it turns out we can't show the video. -->
-    <string name="explore_io_video_id_not_valid">Sorry, there appears to be a problem with the video url.</string>
-
-    <!-- END Explore Message Card User Messages -->
-
-    <!-- This string is the title of a notification shown to the user when Google Play
-        services can't connect. -->
-    <string name="app_data_notification_resolution_title">I/O App needs permission</string>
-    <!-- This string is the message of the notification shown to the user when Google Play services
-         can't connect from a background sync. -->
-    <string name="app_data_notification_resolution_content">I/O App needs action to complete background sync.</string>
-
     <!-- Generic error message used for errors that should never happen. This explains to the user
          that we can't execute the request when they click a button or touch a part of the screen
          that should normally be touchable. -->
@@ -101,4 +38,8 @@
     <string name="show_floating_app_icon_description">Show floating app icon</string>
     <string name="dashboard_question_text">What would you like to do?</string>
     <string name="empty_tutorial">Oops! No content here. Please try the next tutorial.</string>
+    <string name="game_over_text">Game Over!</string>
+    <string name="play_again_text">Play Again</string>
+    <string name="quiz_template_points"><xliff:g id="current_step_number">%1$d</xliff:g> Points</string>
+    <string name="placeholder_quiz_score">Score: 0</string>
 </resources>

